tests/testthat/test_extractData2.R

```diff
@@ -68,16 +69,48 @@ test_that("Extract data into factor with duplicate value labels", {
   expect_equal(out2$VAR1, out_factor2)
 })
 
+test_that("Extract data into factor with duplicate value labels but only on missings", {
+  testM3 <- changeValLabels(testM2, varName = "VAR1", value = "-99", valLabel = "Omission")
+  outW <- capture_warnings(out <- extractData2(testM3, labels2factor = "VAR1", convertMiss = TRUE, dropPartialLabels = FALSE))
+
+  expect_equal(length(outW), 1)
+  expect_equal(class(out$VAR1), "factor")
+  out_factor <- factor(c("One", NA, NA, "2"))
+  attr(out_factor, "label") <- "Variable 1"
+  expect_equal(out$VAR1, out_factor)
+
+  testM4 <- changeValLabels(testM2, varName = "VAR1", value = "-99", valLabel = "Omission")
+  testM4 <- changeValLabels(testM4, varName = "VAR1", value = "2", valLabel = "Two")
+  out2 <- extractData2(testM4, labels2factor = "VAR1", convertMiss = TRUE)
+
+  expect_equal(class(out2$VAR1), "factor")
+  out_factor2 <- factor(c("One", NA, NA, "Two"))
+  attr(out_factor2, "label") <- "Variable 1"
+  expect_equal(out2$VAR1, out_factor2)
+})
+
+test_that("Extract data into factor with one value label being NA", {
+  testM3 <- changeValLabels(testM2, varName = "VAR1", value = 2, valLabel = NA)
+
+  out <- extractData2(testM3, labels2character = "VAR1", convertMiss = TRUE)
+  expect_equal(as.character(out$VAR1), c("One", NA, NA, NA))
+
+  outF <- extractData2(testM3, labels2factor = "VAR1", convertMiss = TRUE)
+  fac <- factor(c(1, NA, NA, 2), levels = 1:2, labels = c("One", NA))
+  attr(fac, "label") <- "Variable 1"
+  expect_equal(outF$VAR1, fac)
+})
+
 test_that("Extract data into factor with multiple value labels being NA", {
```

Comment:
Doesn't the first test check even the more general scenario of one or more value labels being `NA` while the second one tests the special case of *all* value labels being `NA`? Or is the crucial difference actually between only one and multiple `NA`s?

Reply:
If have adapted the name of the test. Do you think this is sufficient or should a test for "multiple but not all value labels are `NA`" be added?

Reply:
I am somewhat unsure because I also don't have a great overview of everything that should and could happen during ´extractData2´. I could imagine testing the function on marginally larger test datasets and specifically differentiating "multiple but not all" from "all" may be a good idea for the future.